packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,6 +206,46 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(
+    { body }: TxInternals,
+    { inputAddressResolver }: SignTransactionOptions
+  ): Promise<Cardano.Signatures> {
+    try {
+      const { accountIndex, networkId, communicationType, knownAddresses } = this
+        .serializableData as SerializableLedgerKeyAgentData;
```

Comment:
These are already available as `this.member`. You should probably create a new `this.#communicationType` setting it in the constructor.